app/views/publishers/home.html.slim

```diff
@@ -139,56 +140,58 @@ javascript:
-
-            statementDownloadDiv = document.createElement('div');
-            statementDownloadDiv.className = 'download';
-            statementDownloadDiv.appendChild(document.createTextNode('Generating'));
-            newStatementDiv.appendChild(statementDownloadDiv);
-
-            generatedStatements.insertBefore(newStatementDiv, generatedStatements.firstChild);
-
-            dynamicEllipsis.start(statementDownloadDiv);
+          event.preventDefault();
+          generateStatement.style.display = 'none';
 
-            statementId = json.id;
-            return pollUntilSuccess('/publishers/statement_ready?id=' + statementId, 3000, 2000, 7);
-          })
-          .then(function() {
-            dynamicEllipsis.stop(statementDownloadDiv);
-            statementDownloadDiv.innerHTML = '<a href="/publishers/statement?id=' + statementId + '">Download</a>';
-            generateStatement.style.display = 'inline-block';
-          })
-          .catch(function(e) {
-            if (statementDownloadDiv) {
+          submitForm('statement_generator', 'PATCH', false)
+            .then(function(response) {
+              return response.json();
+            })
+            .then(function(json) {
+              statementPeriod.options.remove(statementPeriod.selectedIndex);
+              if (statementPeriod.options.length === 0) {
+                statementGenerator.style.display = 'none';
+              }
+
+              var newStatementDiv = document.createElement('div');
+              newStatementDiv.className = 'statement';
+
+              var statementPeriodDiv = document.createElement('div');
+              statementPeriodDiv.className = 'period';
+              statementPeriodDiv.appendChild(document.createTextNode(json.period));
+              newStatementDiv.appendChild(statementPeriodDiv);
+
+              statementDownloadDiv = document.createElement('div');
+              statementDownloadDiv.className = 'download';
+              statementDownloadDiv.appendChild(document.createTextNode('Generating'));
+              newStatementDiv.appendChild(statementDownloadDiv);
+
+              generatedStatements.insertBefore(newStatementDiv, generatedStatements.firstChild);
+
+              dynamicEllipsis.start(statementDownloadDiv);
+
+              statementId = json.id;
+              return pollUntilSuccess('/publishers/statement_ready?id=' + statementId, 3000, 2000, 7);
+            })
+            .then(function() {
               dynamicEllipsis.stop(statementDownloadDiv);
-              statementDownloadDiv.innerText = 'Delayed'
-            }
-            generateStatement.style.display = 'inline-block';
-          });
-      }, false);
+              statementDownloadDiv.innerHTML = '<a href="/publishers/statement?id=' + statementId + '">Download</a>';
+              generateStatement.style.display = 'inline-block';
+            })
+            .catch(function(e) {
```

Comment:
Is this `catch` expecting that the error is a timeout? Could you check to be sure the error is what you expect for this logic, and propagate if it is unexpected? For example:

```js
.catch(function(e) {
  if (e instanceof TimeoutError === false) {
    // Error is unexpected, propagate
    throw e;
  }
  // Logic to handle the timeout case
});
```

This will keep the code from swallowing unexpected errors, something that makes it difficult to debug unexpected faults later on.

Reply:
This seems reasonable; let's feel free to do a little patch after this one. Thanks!

Reply:
Yep, I agree. Tracking this in #352.